File: frontend/src/services/recruitmentService.js
import apiClient from './api';

export const getJobs = () => {
  return apiClient.get('/recruitment/jobs');
};

export const createJob = (dto) => {
  return apiClient.post('/recruitment/jobs', dto);
};

export const getCandidates = () => {
  return apiClient.get('/recruitment/candidates');
};

export const updateCandidateStage = (id, dto) => {
  return apiClient.put(`/recruitment/candidates/${id}/stage`, dto);
};

export const deleteCandidate = (id) => {
  return apiClient.delete(`/recruitment/candidates/${id}`);
};

export const applyToJob = (dto) => {
  return apiClient.post('/recruitment/apply', dto);
};

File: backend/src/main/java/com/awais/hr/module/recruitment/service/RecruitmentService.java
package com.awais.hr.module.recruitment.service;

import com.awais.hr.module.recruitment.dto.CandidateStageUpdateDTO;
import com.awais.hr.module.recruitment.dto.JobRequisitionRequestDTO;
import java.util.List;
import java.util.Map;

public interface RecruitmentService {
    List<Map<String, Object>> getJobs();
    void createJob(JobRequisitionRequestDTO dto);
    List<Map<String, Object>> getCandidates(String email);
    void updateCandidateStage(String id, CandidateStageUpdateDTO dto);
    void deleteCandidate(String id);
    void applyToJob(Map<String, String> application);
}
